backend/src/types.ts

```diff
@@ -11,12 +11,15 @@ type BoardState = {
   players: string[];
 };
 
-interface HandlerParams {
+interface BaseHandlerParams {
   socket: Socket;
   io: Server;
-  username: string;
-  rooms: Map<string, RoomData>;
+  roomsMap: Map<string, RoomData>;
   boards: Map<string, BoardState>;
 }
 
-export { RoomData, BoardState, HandlerParams };
+interface HandlerParams extends BaseHandlerParams {
```

Comment:
Are we just using this for login? If so, we should name it something like LoginParams instead of HandlerParams. HandlerParms seems generic but the BaseHandlerParams is the generic.

Reply:
We use HandlerParams for everything _except_ login, since it just adds a username field to BaseHandlerParams. So it should be generic enough to apply to all the other handlers. It does still seem kinda generic but idk what other name could work.